docs/developer/make_verify.md

```diff
@@ -0,0 +1,235 @@
+    &emsp;3.then comparing the original code with the freshly generated code.   
+If there are differences, it prompts the user to update the code generation by running **`hack/update-codegen.sh`**. This helps maintain the codebase consistency and correctness.
+
+## 5.Verify vendor licenses 
+
+The sub-target **`verify-vendor-licenses`** will run its corrsponding shell script
+
+```shell
+verify-vendor-licenses:
+    hack\verify-vendor-licenses.sh
+```
+
+### Verify licenses steps
+
+**5.1**. Defining Variables and clean up environment
+
+```shell
+KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+source "${KUBE_ROOT}/hack/lib/init.sh"
+kube::util::trap_add cleanup EXIT
+```
+
+**5.2**. Run **`update-vendor-licenses.sh`** script 
+
+```shell
+LICENSE_ROOT="${_tmpdir}" "${KUBE_ROOT}/hack/update-vendor-licenses.sh"
+```
+
+**5.3**. Comparing licenses directory
+
+```shell
+  if ! _out="$(diff -Naupr -x OWNERS "${KUBE_ROOT}/LICENSES" "${_tmpdir}/LICENSES")"; then
+      echo "Your LICENSES tree is out of date. Run hack/update-vendor-licenses.sh and commit the results." >&2
+  fi
+```
+
+This script ensure that the licenses in kubeedge are up-to-date by generating new licenses and comparing them with
+the current ones.  
+It creates a temporary workspace, symlinks, necessary directoires and files, and then compares.  
+If there are differences, it prompts the user to update the licenses by running **`hack/update-vendor-licenses.sh`**
+
+## 6.Verify crds
+
+The sub-target **`verify-crds`** will run its corrsponding shell script
+
+```shell
+verify-crds:
+    hack\verify-crds.sh
+```
+
+### Verify CRDs steps
+
+**6.1**. Defining Variables and clean up environment
+
+```shell
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE}")/..
+DIFFROOT="${SCRIPT_ROOT}/build/crds"
+TMP_DIFFROOT="${SCRIPT_ROOT}/_tmp/build/crds"
+_tmp="${SCRIPT_ROOT}/_tmp"
+cleanup
+```
+
+**6.2**. Run **`generate-crds.sh`**
+
+```shell
+"${SCRIPT_ROOT}/hack/generate-crds.sh"
+```
+
+**6.3**. Comparing crds directory and update original directory
+
+```shell
+diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
+```
+
+**6.4**. Check differences.
+
+
+This script ensure that the Custom Resource Difinitions (CRDs) in the build/crds directory are up-to-date by comparying
+them with freshly generated CRDs.  
+If there are differences, it prompts the user to update the CRDs by running
+**`hack/generate-crds.sh`**
+
+## Summary 
+
+When user run **`make verify`**, all the verification scripts will be executed in sequence. These scripts usually include various
+checks on the project's code, dependencies, generated code, and more, to ensure code quality and consistency.  
+If the **HELP** variable is set to **y**, these verification tasks won't be executed, and instead, a help message will be printed to display
+```shell
+# verify golang, vendor, codegen and crds
+#
+# Example:
+# make verify
+```
```

Comment:
The code here doesn't seem to make much sense, would you consider deleting it?

Reply:
Removed this part.